lib/puppet/type/file/content.rb

```diff
@@ -220,6 +221,19 @@ def get_from_source(source_or_content, &block)
       end
     end
 
+    def get_from_http_source(source, &block)
+      Puppet::Util::HttpProxy.request_with_redirects(URI(source), :get, &block)
```

Comment:
Come to think of it, it would appear that I should just rely on the [option](https://github.com/puppetlabs/puppet/blob/a9a0e9abdee14931966fe036fc723062dca66dd5/lib/puppet/network/http/connection.rb#L59) for that in the first place, and not expose it as a parameter at all.